src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -234,6 +233,10 @@ private async Task ExecuteRulesAsync(IEnumerable<AsyncConsensusRule> asyncRules,
             {
                 ruleContext.ValidationContext.Error = ex.ConsensusError;
             }
+            catch (Exception ex)
+            {
+                ruleContext.ValidationContext.Error = new ConsensusError("unhandled-exception", ex.Message);
```

Comment:
Could you do `ex.ToString()` here ? As there's an override in `ConsensusError` which includes the error code, as well as the message.

Reply:
Done :)